app/controllers/LongerCommodityCodeController.scala

```diff
@@ -137,26 +131,26 @@ class LongerCommodityCodeController @Inject() (
                                    Some(recordId)
                                  )
       updatedAnswers          <- Future.fromTry(request.userAnswers.set(LongerCommodityCodePage(recordId), value))
-      updatedAnswersWithQuery <- Future.fromTry(updatedAnswers.set(LongerCommodityQuery(recordId), validCommodityCode))
+      cleansedAnswers         <- if (shouldRedirect) {
+                                   Future.successful(updatedAnswers)
+                                 } else {
+                                   Future.fromTry(updatedAnswers.remove(HasCorrectGoodsLongerCommodityCodePage(recordId)))
```

Comment:
i think this should be in the LongerCommodityCodePage file - if you check the one for the plain commodityCodePage you can see it has a cleanup for this

Reply:
Agreed

Reply:
Actually, cleanup does not support recordId and that is needed for the cleanup method to reference the pervious value... had to undo the change...

Reply:
Cleanup doesn't need to pass in recordId, it is in scope as its in the Page definition. 

See e.g. HasSupplementaryUnitPage cleanup

Reply:
I'll take a look now, thanks chris, I haven't had a chance yet but this morning I thought I'd probably been too pessimistic here